src/main/java/com/sourcegraph/cody/CodyToolWindowContent.kt

```diff
@@ -84,72 +73,12 @@ class CodyToolWindowContent(private val project: Project) : UpdatableChat {
     sendButton = createSendButton(project)
-    val autoGrowingTextArea = AutoGrowingTextArea(3, 9, promptPanel)
-    promptInput = autoGrowingTextArea.textArea
-    /* Submit on enter */
-    val JUST_ENTER = KeyboardShortcut(getKeyStroke(KeyEvent.VK_ENTER, 0), null)
-    val UP = KeyboardShortcut(getKeyStroke(KeyEvent.VK_UP, 0), null)
-    val DOWN = KeyboardShortcut(getKeyStroke(KeyEvent.VK_DOWN, 0), null)
-    val DEFAULT_SUBMIT_ACTION_SHORTCUT = CustomShortcutSet(JUST_ENTER)
-    val POP_UPPER_MESSAGE_ACTION_SHORTCUT = CustomShortcutSet(UP)
-    val POP_LOWER_MESSAGE_ACTION_SHORTCUT = CustomShortcutSet(DOWN)
-    val upperMessageAction: AnAction =
-        object : DumbAwareAction() {
-          override fun actionPerformed(e: AnActionEvent) {
-            if (isInHistoryMode) {
-              chatMessageHistory.popUpperMessage(promptInput)
-            } else {
-              val defaultAction = promptInput.actionMap[DefaultEditorKit.upAction]
-              defaultAction.actionPerformed(null)
-            }
-          }
-        }
-    val lowerMessageAction: AnAction =
-        object : DumbAwareAction() {
-          override fun actionPerformed(e: AnActionEvent) {
-            if (isInHistoryMode) {
-              chatMessageHistory.popLowerMessage(promptInput)
-            } else {
-              val defaultAction = promptInput.actionMap[DefaultEditorKit.downAction]
-              defaultAction.actionPerformed(null)
-            }
-          }
-        }
-    val sendMessageAction: AnAction =
-        object : DumbAwareAction() {
-          override fun actionPerformed(e: AnActionEvent) {
-            if (promptInput.getText().isNotEmpty()) {
-              sendChatMessage(project)
-            }
-          }
-        }
-    sendMessageAction.registerCustomShortcutSet(DEFAULT_SUBMIT_ACTION_SHORTCUT, promptInput)
-    upperMessageAction.registerCustomShortcutSet(POP_UPPER_MESSAGE_ACTION_SHORTCUT, promptInput)
-    lowerMessageAction.registerCustomShortcutSet(POP_LOWER_MESSAGE_ACTION_SHORTCUT, promptInput)
-    promptInput.addKeyListener(
-        object : KeyAdapter() {
-          override fun keyReleased(e: KeyEvent) {
-            val keyCode = e.keyCode
-            if (keyCode != KeyEvent.VK_UP && keyCode != KeyEvent.VK_DOWN) {
-              isInHistoryMode = promptInput.getText().isEmpty()
-            }
-          }
-        })
-    // Enable/disable the send button based on whether promptInput is empty
-    promptInput.document.addDocumentListener(
-        object : DocumentAdapter() {
-          override fun textChanged(e: DocumentEvent) {
-            sendButton.isEnabled = promptInput.getText().isNotEmpty()
-          }
-        })
-    promptPanel.add(autoGrowingTextArea.scrollPane, BorderLayout.CENTER)
-    promptPanel.border = BorderFactory.createEmptyBorder(0, 0, 10, 0)
+    val promptPanel =
+        PromptPanel(project, chatMessageHistory, ::sendChatMessage) { sendButton.isEnabled = it }
```

Comment:
IMO there is small issue with readability - I had to check the implementation of the constructor to make sure what we are doing with this trailing lambda.

Suggestion: we should use named parameter here, like `onTextChanged = { sendButton.isEnabled = it }`